nx/lib/nx/defn/expr.ex

```diff
@@ -637,10 +646,21 @@ defmodule Nx.Defn.Expr do
     else
       [s | start] = start
       [l | lengths] = lengths
-      [{is + s, l} | merge_slice(axis + 1, axes, inner_start, start, inner_lengths, lengths)]
+
+      [
+        {Nx.add(is, s), l}
```

Comment:
This needs to keep them integers if they are integers. I think kernel.+ will do it:

```suggestion
        {Nx.Defn.Kernel.+(is, s), l}
```